### PR TITLE
feat: handle long options better

### PR DESCRIPTION
**Pull request by robbat2**
- fix: do not die on long DNSSL option
- feat: improve warning about long RA options

See more detail in individual commit messages.

Reference: https://datatracker.ietf.org/doc/html/rfc8106#section-5.2
 Reference: https://datatracker.ietf.org/doc/html/rfc6980#section-2
Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

### CODE REVIEW

File: send.c [hunk 1]
```diff
@@ -652,9 +652,14 @@ static struct safe_buffer_list *add_ra_options_dnssl(struct safe_buffer_list *sb
...
 		size_t const bytes = sizeof(dnsslinfo) + domain_name_bytes;
-		if (bytes > (256 * 8)) {
-			flog(LOG_ERR, "DNSSL too long for RA option, must be < 2048 bytes.  Exiting.");
-			exit(1);
+		// dnsslinfo.nd_opt_dnssli_len is uint8 count of 8-octet groups; min 3, max 255
+		// too many long serialized domains could exceed it
+		// https://datatracker.ietf.org/doc/html/rfc8106#section-5.2
+		size_t const bytes_rounded_up = (bytes + 7) / 8;
+		if(bytes_rounded_up > (255*8)) {
+			flog(LOG_ERR, "DNSSL too long for RA option, must be < 2032 bytes. Skipping option.");
```
Comment (mpontillo) [inline send.c:660]: I'm confused about the math here. `255*8` is `2040`. So should this error message read `<= 2040 bytes`? Or did I miss something?

Comment (SpareSimian) [reply send.c:660]: A lot of magic numbers here. I suggest adding macro constants to document what they mean.

Comment (robbat2) [reply send.c:660]: 2032 bytes of suffixes, plus 8 bytes of header = 2040; will introduce macros for clarity.

But I also think there is a subtle bug here with very large configs that needs fixing

Comment (robbat2) [reply send.c]: @mpontillo  this should be much clearer now - I also added testcases for the edges involved

**Comment by robbat2** [comment]
I want to merge this before #245

**Comment by stappersg** [comment]
On Sun, Dec 29, 2024 at 05:30:53PM -0800, Robin H. Johnson wrote:
> I want to merge this before #245  


Fine for me.

However when this email was written was #245 merged
and this merge request not.


For the sake of moving forward: I'm fine with moving forward.
And I'm find with a revert of #245 and #244, this MR, going first.

I'll aprove this MR so there is no blocker.



Something else: I'm somewhat worried how this merge request,
which consists of several commits, will land in the radvd git repo.
(I fear an ugly commit)


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Review by stappersg — APPROVED** [review]
Approved for _moving on_.
